coffea/nanoevents/schemas/physlite.py

```diff
@@ -17,31 +19,87 @@ class PHYSLITESchema(BaseSchema):
+    cross_reference_elementlinks = {
+        ("Electrons", "trackParticleLinks"): "GSFTrackParticlesAuxDyn.z0",
+    }
+
     def __init__(self, base_form):
         super().__init__(base_form)
         self._form["contents"] = self._build_collections(self._form["contents"])
 
     def _build_collections(self, branch_forms):
         zip_groups = defaultdict(list)
         for key, ak_form in branch_forms.items():
+            # Normal fields
             key_fields = key.split("/")[-1].split(".")
             top_key = key_fields[0]
             sub_key = ".".join(key_fields[1:])
             objname = top_key.replace("Analysis", "").replace("AuxDyn", "")
-            zip_groups[objname].append((key, sub_key))
+            zip_groups[objname].append(((key, sub_key), ak_form))
+
+            # Global indices
+            for cross_references in [
+                self.cross_reference_indices,
+                self.cross_reference_elementlinks,
+            ]:
+                if (objname, sub_key) in cross_references:
+                    linkto_key = cross_references[(objname, sub_key)]
+                    if cross_references is self.cross_reference_indices:
+                        form = self._create_global_index_form(ak_form, key, linkto_key)
+                    elif cross_references is self.cross_reference_elementlinks:
+                        form = self._create_global_index_form_elementlink(
+                            ak_form, key, linkto_key
+                        )
+                    zip_groups[objname].append(((key, sub_key + "G"), form))
 
         contents = {}
-        for objname, keys in zip_groups.items():
+        for objname, keys_and_form in zip_groups.items():
             try:
                 contents[objname] = zip_forms(
-                    {sub_key: branch_forms[key] for key, sub_key in keys},
+                    {sub_key: form for (key, sub_key), form in keys_and_form},
                     objname,
                     self.mixins.get(objname, None),
                 )
             except NotImplementedError:
                 warnings.warn(f"Can't zip collection {objname}")
         return contents
 
+    @staticmethod
+    def _create_global_index_form(base_form, key, linkto_key):
+        form = copy.deepcopy(base_form)
+        form["content"]["form_key"] = quote(
+            f"{key},!load,{linkto_key},!load,!offsets,!to_numpy,!local2global"
+        )
+        form["content"]["itemsize"] = 8
+        form["content"]["primitive"] = "int64"
+        return form
+
+    @staticmethod
+    def _create_global_index_form_elementlink(base_form, key, linkto_key):
+        form = copy.deepcopy(base_form)
+        record = form["content"]["content"]["contents"]
+        form["content"]["content"] = record["m_persIndex"]
+        form["content"]["content"]["form_key"] = quote(
+            f"{key},!load,m_persIndex,!item,{linkto_key},!load,!offsets,!to_numpy,!local2global"
```

Comment:
for the double-jagged  cross references i have to pop off the `m_persKey` since these are the branches that can't be split